go/apps/http_api_nostream/vumi_app.py

```diff
@@ -326,7 +326,12 @@ def schedule_push_retry(self, user_account_key, url, method, data,
             resp = yield http_request_full(
                 retry_url, data=retry_data, headers=retry_headers,
                 timeout=self.http_retry_timeout)
-            if not (200 <= resp.code < 300):
+            if resp.code == 429:
+                log.warning(
+                    "Retrying is rate limited for account '%s'."
+                    " Discarding HTTP request." % (user_account_key,))
+                returnValue(None)
```

Comment:
One can have a `return` without an expression in a generator, and for some reason I thought we use that below `returnValue`s to short circuit, but I can't seem to find any place in the codebase where we do that, so I must be remembering wrong. Sorry.